src/Compilers/Core/MSBuildTask/ValidateBootstrap.cs

```diff
@@ -87,17 +87,18 @@ public override bool Execute()
             return allGood;
         }
 
-        private static string NormalizePath(string path)
+        private static string? NormalizePath(string? path)
         {
+            // PROTOTYPE(NullableDogfood): string.IsNullOrEmpty needs annotation
             if (string.IsNullOrEmpty(path))
             {
                 return path;
             }
 
-            var c = path[path.Length - 1];
+            var c = path![path!.Length - 1];
             if (c == Path.DirectorySeparatorChar || c == Path.AltDirectorySeparatorChar)
             {
-                path = path.Substring(0, path.Length - 1);
+                path = path!.Substring(0, path!.Length - 1);
```

Comment:
❔ Is there a reason `!` doesn't propagate in data flow analysis?

Reply:
I don't remember the rationale.